### PR TITLE
replace the AddToCalendar package with our own functionality

### PR DESCRIPTION
**Pull request by s-cheng**
Since the "add to calendar" functionality [stopped working](https://github.com/ForumMagnum/ForumMagnum/issues/4851), and I wanted to replace the icon eventually anyway, I decided to look into building our own thing.

I borrowed a bunch from the existing package, and it turned out to be a relatively small amount of code. I also made an icon by manually combining two of MUI's icons to make this one (which they really should already have).

Screen Shot 2022-05-03 at 2 02 41 PM

------
_Bonus for the EA Forum_: Now that the icon is no longer a frankenstein of parts with separate backgrounds, I can put this component in places where the background is not white! 

Screen Shot 2022-05-03 at 2 12 32 PM

### CODE REVIEW

File: packages/lesswrong/components/form-components/LocationFormComponent.tsx [hunk 1]
```diff
@@ -128,10 +128,6 @@ export const useGoogleMaps = (): [boolean, any] => {
 }
 
 
-const getLocationString = (googleLocation: any): string => {
-  return googleLocation?.["formatted_address"];
-}
```
Comment (s-cheng) [inline packages/lesswrong/components/form-components/LocationFormComponent.tsx:133]: This was unused

File: packages/lesswrong/components/posts/AddToCalendar/AddToCalendarButton.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+})
+
+const AddToCalendarIcon = ({className=''}) => {
+  return <svg viewBox="207.59 11.407 18.981 20.638" xmlns="http://www.w3.org/2000/svg" className={className}>
+    <path d="M 225.59 21.407 L 225.59 15.407 C 225.59 14.307 224.69 13.407 223.59 13.407 L 222.59 13.407 L 222.59 11.407 L 220.59 11.407 L 220.59 13.407 L 212.59 13.407 L 212.59 11.407 L 210.59 11.407 L 210.59 13.407 L 209.59 13.407 C 208.49 13.407 207.59 14.307 207.59 15.407 L 207.59 29.407 C 207.59 30.507 208.49 31.407 209.59 31.407 L 216.59 31.407 L 216.59 29.407 L 209.59 29.407 L 209.59 19.407 L 223.59 19.407 L 223.59 21.407 L 225.59 21.407 Z"/>
+    <path d="M 226.571 27.045 L 226.571 29.045 L 223.571 29.045 L 223.571 32.045 L 221.571 32.045 L 221.571 29.045 L 218.571 29.045 L 218.571 27.045 L 221.571 27.045 L 221.571 24.045 L 223.571 24.045 L 223.571 27.045 L 226.571 27.045 Z"/>
+  </svg>
+}
+
+const AddToCalendarButton = ({post, label, hideTooltip, hideIcon, iconClassName, classes}: {
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/AddToCalendar/AddToCalendarButton.tsx:56]: I renamed this from `AddToCalendarIcon` - most of this component is the same as before

File: packages/lesswrong/components/posts/AddToCalendar/AddToCalendarButton.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    <div className={classes.root}>
+      <button className={classes.button} onClick={handleClick}>
+        {!hideIcon && <AddToCalendarIcon className={classNames(classes.icon, iconClassName)} />}
+        {label && (
+          <span className={classes.label}>
+            {label}
+          </span>
+        )}
+      </button>
+      {open && <div className={classes.dropdown}>
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/AddToCalendar/AddToCalendarButton.tsx:128]: Another bonus, I got to reorder the dropdown options based on usage (so Google is first)

File: packages/lesswrong/components/posts/AddToCalendar/makeUrls.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
+/**
+ * This file is copied from https://github.com/CultureHQ/add-to-calendar/blob/master/src/makeUrls.ts
+ * (I deleted Yahoo since no one uses that)
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/AddToCalendar/makeUrls.ts:3]: Add their license:

https://github.com/CultureHQ/add-to-calendar/blob/master/LICENSE

**Review by jpaddison3 — APPROVED** [review]

**Comment by jpaddison3** [comment]
I love the bonuses.

Another bonus: It now will work with dark theme, whereas before it was gonna need work.

**Pull request merged**